### PR TITLE
Documentation about User Access Tokens

### PR DESCRIPTION
**Pull request by SBrandeis**
Documentation about the new User Access Tokens

Suggestions and feedback more than  welcome! 🤗 

image

### CODE REVIEW

File: docs/hub/security.md [hunk 1]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+User Access Tokens are the preferred way to authenticate an application or a notebook to the HuggingFace services. You can manage your access tokens in your [settings](https://huggingface.co/settings/tokens).
+
+![/docs/assets/hub/access-tokens.png](/docs/assets/hub/access-tokens.png)
+
+Access tokens allow applications and notebooks to impersonate you, with restricted permissions. Each token is given a role, among the following:
+
+- `read`: tokens with this role can only be used to provide read access to repositories you could read. That includes public repositories and private repositories that you or an org you're a member of own. Use this role if you only need to read content from the HuggingFace hub (eg, when doing inference).
+
+- `write`: tokens with this role additionally grant write access to the repositories you have write access to. Use this token if you need to push content to a repository (eg, when training a model).
+
```
Comment (sgugger) [inline docs/hub/security.md:20]: No doc for the admin token?

File: docs/hub/security.md [hunk 2]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+User Access Tokens are the preferred way to authenticate an application or a notebook to the HuggingFace services. You can manage your access tokens in your [settings](https://huggingface.co/settings/tokens).
+
+![/docs/assets/hub/access-tokens.png](/docs/assets/hub/access-tokens.png)
+
+Access tokens allow applications and notebooks to impersonate you, with restricted permissions. Each token is given a role, among the following:
+
+- `read`: tokens with this role can only be used to provide read access to repositories you could read. That includes public repositories and private repositories that you or an org you're a member of own. Use this role if you only need to read content from the HuggingFace hub (eg, when doing inference).
+
+- `write`: tokens with this role additionally grant write access to the repositories you have write access to. Use this token if you need to push content to a repository (eg, when training a model).
```
Comment (sgugger) [inline docs/hub/security.md:19]:
```suggestion
- `write`: tokens with this role additionally grant write access to the repositories you have write access to. Use this token if you need to create or push content to a repository (eg, when training a model).
```

File: docs/hub/security.md [hunk 3]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+access_token = "hf_..."
+
+model = AutoModel.from_pretrained("private/model", use_auth_token=access_token)
+```
+
+### Best practices
+
+We recommand to create one access token per app or usage. For instance, one would have one token to use on their local machine, one token to use on Colab notebooks, and one token to be used by an awesome custom inference server. This way, you can invalidate one token without impacting your other usages.
+
+We also recommand giving the minimal rights to each token you create. If you only need read access (ie, loading a dataset with the `datasets` library or retrieving the weights of a model), give your access token the `read` role.
```
Comment (sgugger) [inline docs/hub/security.md:53]:
```suggestion
We also recommand giving the minimal rights to each token you create. If you only need read access (i.e., loading a dataset with the `datasets` library or retrieving the weights of a model), give your access token the `read` role.
```

**Review by sgugger — COMMENTED** [review]
LGTM! Thanks for adding this documentation!

File: docs/hub/security.md (hunk 1 above)
Comment (SBrandeis) [reply docs/hub/security.md:20]: We are discussing wether or not we want to keep it, as it's essentially useless now - but could grant programmatic access to organizations' settings in the future

File: docs/hub/security.md [hunk 4]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
 
 # Security and trust on the Hugging Face Hub
 
+## User Access Tokens
+
+### What are User Access Tokens?
+
+User Access Tokens are the preferred way to authenticate an application or a notebook to the HuggingFace services. You can manage your access tokens in your [settings](https://huggingface.co/settings/tokens).
```
Comment (osanseviero) [inline docs/hub/security.md:11]: This link is a 404

```suggestion
User Access Tokens are the preferred way to authenticate an application or a notebook to the HuggingFace services. You can manage your access tokens in your [settings](https://huggingface.co/settings/token).
```

File: docs/hub/security.md [hunk 5]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+
+![/docs/assets/hub/access-tokens.png](/docs/assets/hub/access-tokens.png)
+
+Access tokens allow applications and notebooks to impersonate you, with restricted permissions. Each token is given a role, among the following:
```
Comment (osanseviero) [inline docs/hub/security.md:15]:
```suggestion
Access tokens allow applications and notebooks to impersonate you with restricted permissions. Each token is given a role, among the following:
```

File: docs/hub/security.md [hunk 6]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+
+- `read`: tokens with this role can only be used to provide read access to repositories you could read. That includes public repositories and private repositories that you or an org you're a member of own. Use this role if you only need to read content from the HuggingFace hub (eg, when doing inference).
```
Comment (osanseviero) [inline docs/hub/security.md:17]:
```suggestion
- `read`: tokens with this role can only be used to provide read access to repositories you could read. That includes public repositories and private repositories that you or an org you're a member of own. Use this role if you only need to read content from the Hugging Face Hub (e.g. when downloading private models or doing inference).
```

File: docs/hub/security.md [hunk 7]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+
+- `write`: tokens with this role additionally grant write access to the repositories you have write access to. Use this token if you need to create or push content to a repository (eg, when training a model).
```
Comment (osanseviero) [inline docs/hub/security.md:19]:
```suggestion
- `write`: tokens with this role additionally grant write access to the repositories you have write access to. Use this token if you need to create or push content to a repository (e.g., when training a model or modifying a model card).
```

File: docs/hub/security.md [hunk 8]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+
+Select a role and a name for your token and voilà - you're ready to go!
+
+You can delete an refresh User Access Tokens by clicking on the "Manage" button.
+
+![/docs/assets/hub/delete-token.png](/docs/assets/hub/delete-token.png)
+
+### How to use User Access Tokens?
+
+There are plenty of ways to use a User Access Token to access the Hugging Face hub, granting you the flexibility you need to build awesome apps on top of it.
```
Comment (osanseviero) [inline docs/hub/security.md:35]:
```suggestion
There are plenty of ways to use a User Access Token to access the Hugging Face Hub, granting you the flexibility you need to build awesome apps on top of it.
```

File: docs/hub/security.md [hunk 9]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+
+- User Access Tokens can be used **in place of a password** to access the Hugging Face hub with git or with Basic authentication
```
Comment (osanseviero) [inline docs/hub/security.md:37]:
```suggestion
- User Access Tokens can be used **in place of a password** to access the Hugging Face Hub with git or with basic authentication
```

File: docs/hub/security.md [hunk 10]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+- User Access Tokens can be passed as a **bearer token** when calling the Hugging Face APIs
```
Comment (osanseviero) [inline docs/hub/security.md:38]:
```suggestion
- User Access Tokens can be passed as a **bearer token** when calling the [Inference API](https://huggingface.co/inference-api).
```

File: docs/hub/security.md [hunk 11]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+- User Access Tokens can be used in the Hugging Face Python library:
```
Comment (osanseviero) [inline docs/hub/security.md:39]: Not 100% sure we should link to that page

```suggestion
- User Access Tokens can be used in the `huggingface_hub` [Python library](https://github.com/huggingface/huggingface_hub/tree/main/src/huggingface_hub):
```

**Review by osanseviero — APPROVED** [review]
This is great! Thanks a lot. I left a couple of minor suggestions and improvements, but overall LGTM

File: docs/hub/security.md (hunk 4 above)
Comment (stevhliu) [inline docs/hub/security.md]:
```suggestion
User Access Tokens are the preferred way to authenticate an application or notebook to Hugging Face services. You can manage your access tokens in your [settings](https://huggingface.co/settings/tokens).
```

File: docs/hub/security.md (hunk 5 above)
Comment (stevhliu) [inline docs/hub/security.md]:
```suggestion
Access tokens allow applications and notebooks to perform specific actions specified by the scope of the roles shown in the following:
```

File: docs/hub/security.md (hunk 6 above)
Comment (stevhliu) [inline docs/hub/security.md]:
```suggestion
- `read`: tokens with this role can only be used to provide read access to repositories you could read. That includes public repositories and private repositories that you, or an organization you're a member of, own. Use this role if you only need to read content from the HuggingFace hub (eg, when doing inference).
```

File: docs/hub/security.md [hunk 12]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+
+### How to manage User Access Tokens?
+
+To create an access token, go to your settings, then click on the ["Access Tokens" tab](https://huggingface.co/settings/token). Click on the "New token" button to create a new User Access Token.
+
+![/docs/assets/hub/new-token.png](/docs/assets/hub/new-token.png)
+
+Select a role and a name for your token and voilà - you're ready to go!
+
+You can delete an refresh User Access Tokens by clicking on the "Manage" button.
```
Comment (stevhliu) [inline docs/hub/security.md:29]:
```suggestion
You can delete and refresh User Access Tokens by clicking on the "Manage" button.
```

File: docs/hub/security.md (hunk 11 above)
Comment (SBrandeis) [reply docs/hub/security.md:39]: Shoot I wanted to say "libraries", to refer to `datasets`, `transformers` and `huggingface_hub` all at once

File: docs/hub/security.md [hunk 13]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+
+### What are User Access Tokens?
+
+User Access Tokens are the preferred way to authenticate an application or a notebook to the HuggingFace services. You can manage your access tokens in your [settings](https://huggingface.co/settings/token).
+
+![/docs/assets/hub/access-tokens.png](/docs/assets/hub/access-tokens.png)
+
+Access tokens allow applications and notebooks to impersonate you with restricted permissions. Each token is given a role, among the following:
+
+- `read`: tokens with this role can only be used to provide read access to repositories you could read. That includes public repositories and private repositories that you or an org you're a member of own. Use this role if you only need to read content from the Hugging Face Hub (e.g. when downloading private models or doing inference).
```
Comment (julien-c) [inline docs/hub/security.md:17]:
```suggestion
- `read`: tokens with this role can only be used to provide read access to repositories you could read. That includes public repositories and private repositories that you or one of your orgs own. Use this role if you only need to read content from the Hugging Face Hub (e.g. when downloading private models or doing inference).
```
maybe simpler grammatically? not sure

File: docs/hub/security.md [hunk 14]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+- User Access Tokens can be used in the Hugging Face Python libraries, such as `transformers` or `datasets`:
+
+```python
+from transformers import AutoModel
+
+access_token = "hf_..."
+
+model = AutoModel.from_pretrained("private/model", use_auth_token=access_token)
+```
+
```
Comment (julien-c) [inline docs/hub/security.md:48]:
```suggestion

⚠️ Try not to leak your token! Though you can always rotate it, anyone will be able to read or write your private repos in the meantime which is 💩 
```

**Review by julien-c — APPROVED** [review]
🔥

File: docs/hub/security.md [hunk 15]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+### Best practices
+
+We recommand to create one access token per app or usage. For instance, one would have one token to use on their local machine, one token to use on Colab notebooks, and one token to be used by an awesome custom inference server. This way, you can invalidate one token without impacting your other usages.
```
Comment (stevhliu) [inline docs/hub/security.md:52]:
```suggestion
We recommend you create one access token per app or usage. For instance, you could have a separate token for:

 * A local machine.
 * A Colab notebook.
 * An awesome custom inference server. 
 
 This way, you can invalidate one token without impacting your other usages.
```

File: docs/hub/security.md [hunk 16]
```diff
@@ -4,6 +4,54 @@ title: Security and trust on the Hugging Face Hub
...
+
+We also recommand giving the minimal rights to each token you create. If you only need read access (i.e., loading a dataset with the `datasets` library or retrieving the weights of a model), give your access token the `read` role.
```
Comment (stevhliu) [inline docs/hub/security.md:54]:
```suggestion
We also recommend only giving the appropriate role to each token you create. If you only need read access (i.e., loading a dataset with the `datasets` library or retrieving the weights of a model), only give your access token the `read` role.
```

**Review by stevhliu — COMMENTED** [review]
Looks great! 👍

**Pull request merged**